src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -152,6 +167,23 @@ public static bool IsWindowsDesktop
         private static bool? _isNanoServer = null;
         private static bool? _isIoT = null;
         private static bool? _isWindowsDesktop = null;
+
+        private static readonly Lazy<bool> _isStaSupported = new Lazy<bool>(() =>
+        {
+            // See objbase.h
+            const int COINIT_APARTMENTTHREADED = 0x2;
+            const int E_NOTIMPL = unchecked((int)0X80004001);
+            int result = Windows.NativeMethods.CoInitializeEx(IntPtr.Zero, COINIT_APARTMENTTHREADED);
```

Comment:
I wonder if this PInvoke is anymore efficient than spinning up a thread before.  Has any perf testing been done?